lua/lualine/components/filename.lua

```diff
@@ -1,6 +1,31 @@
-local function filename()
-  local data = [[%t %m]]
-  return data
+local function filename(args)
+  -- setting defaults
+  local modified, full_name, relative = true, false, true
+  if args.modified ~= nil then modified = args.modified end
+  if args.full_name ~= nil then full_name = args.full_name end
+  if args.relative ~= nil then relative = args.relative end
+
+  return function()
+    local data
+    if not full_name then
+      data = vim.fn.expand('%:t')
+    elseif relative then
+      data = vim.fn.expand('%')
+    else
+      data = vim.fn.expand('%:p')
+    end
+    if data == '' then
+      data = '[No Name]'
+    elseif vim.fn.winwidth(0) <= 84 or #data > 40 then
+      data = vim.fn.pathshorten(data)
+    end
+
+    if modified then
+      if vim.bo.modified then data = data .. "[+]"
```

Comment:
This is redundant. By this I mean `[+]` etc. Vim already has a char for that, you can check master for reference. Also this variable shouldn't be called modified but `fileinfo` or something like that

Reply:
>By this I mean [+] etc. Vim already has a char for that, you can check master for reference.

I couldn't really understand what you ment by this . if you're talking about '%m' the only reson I've implemented it in lua so genaral upper and lower works.

> Also this variable shouldn't be called modified but fileinfo or something like that

How about file_status insted?

Reply:
> > By this I mean [+] etc. Vim already has a char for that, you can check master for reference.
> 
> I couldn't really understand what you ment by this . if you're talking about '%m' the only reson I've implemented it in lua so genaral upper and lower works.

Fair enough. 

> > Also this variable shouldn't be called modified but fileinfo or something like that
> 
> How about file_status insted?

Yea sounds good to me